hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRenameRequestV1.java

```diff
@@ -108,6 +115,36 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       checkKeyAcls(ozoneManager, volumeName, bucketName, toKeyName,
               IAccessAuthorizer.ACLType.CREATE, OzoneObj.ResourceType.KEY);
 
+      if (isNativeAuthorizerEnabled) {
```

Comment:
Given the new recursive check flat added, can we move this recursion inside the authorizer and only provide the API for the authorizer to get the key list like what we provide for the ranger authorizer?

Reply:
No, prefixManager does not need to be changed under the current model. All the prefix ACLs has been inherited at key creation time. Check does not need to be recursively populated into prefix.